src/main/java/org/folio/circulation/domain/UpdateRequestService.java

```diff
@@ -15,27 +18,39 @@
   private final ClosedRequestValidator closedRequestValidator;
   private final RequestNoticeSender requestNoticeSender;
   private final UpdateItem updateItem;
+  private final EventPublisher eventPublisher;
 
   public UpdateRequestService(RequestRepository requestRepository,
     UpdateRequestQueue updateRequestQueue, ClosedRequestValidator closedRequestValidator,
-    RequestNoticeSender requestNoticeSender, UpdateItem updateItem) {
+    RequestNoticeSender requestNoticeSender, UpdateItem updateItem, EventPublisher eventPublisher) {
 
     this.requestRepository = requestRepository;
     this.updateRequestQueue = updateRequestQueue;
     this.closedRequestValidator = closedRequestValidator;
     this.requestNoticeSender = requestNoticeSender;
     this.updateItem = updateItem;
+    this.eventPublisher = eventPublisher;
   }
 
   public CompletableFuture<Result<RequestAndRelatedRecords>> replaceRequest(
     RequestAndRelatedRecords requestAndRelatedRecords) {
 
-    return closedRequestValidator.refuseWhenAlreadyClosed(requestAndRelatedRecords)
-      .thenApply(r -> r.next(this::removeRequestQueuePositionWhenCancelled))
-      .thenComposeAsync(r -> r.after(requestRepository::update))
-      .thenComposeAsync(r -> r.after(updateRequestQueue::onCancellation))
-      .thenComposeAsync(r -> r.after(updateItem::onRequestCreateOrUpdate))
-      .thenApply(r -> r.next(requestNoticeSender::sendNoticeOnRequestUpdated));
+    Request updated = requestAndRelatedRecords.getRequest();
+
+    return requestRepository.getById(updated.getId())
```

Comment:
Can the original request be saved into the context object or overload method created so there will be no nested blocks.